terminal.cpp

```diff
@@ -174,19 +174,21 @@ void Terminal::putString(QString str, bool unEscape)
         iPtyIFace->writeTerm(str);
 }
 
-void Terminal::keyPress(int key, int modifiers)
+void Terminal::keyPress(int key, int modifiers, QString text)
 {
     QChar c(key);
 
     resetBackBufferScrollPos();
 
+    if ((!text.isEmpty()) && (text[0].unicode() == 0)) text.clear();
```

Comment:
I know only what i observe, text.count() ==1 and text[0].unicode() == 0 was the result for arrows/other non-letter keys... that broke my logic of assuming you can trust isEmpty()...

Maybe that could use a comment, i wrote all of this not expecting to try to upstream, started as a proof-of-concept bugfix idea.